### PR TITLE
Remove support for puppet lower than 4

### PR DESCRIPTION
**Pull request by Dan33l**
#### Pull Request (PR) description
The goal of this work, based on voxpupuli/puppet-virtualbox puppet4 branch, is to propose a PR that can be merged into master branch.

### CODE REVIEW

File: spec/classes/virtualbox_spec.rb [hunk 1]
```diff
@@ -34,22 +34,16 @@
...
       if facts[:osfamily] == 'Debian'
-        context 'with $::puppetversion < 3.0.0' do
-          let(:facts) { facts.merge(puppetversion: '2.7.26') }
 
-          it { is_expected.not_to contain_class('apt') }
-          it { is_expected.not_to contain_apt__source('virtualbox') }
-        end
-
-        unless Puppet::Util::Package.versioncmp(facts[:puppetversion], '3.0.0') == -1
+        unless Puppet::Util::Package.versioncmp(facts[:puppetversion], '4.0.0') == -1
```
Comment (bastelfreak) [inline spec/classes/virtualbox_spec.rb]: we can get rid of this line. Our testmatrix only uses puppet 4.10 and puppet 5.  It is defined in https://github.com/voxpupuli/modulesync_config/blob/master/config_defaults.yml

**Comment by bastelfreak** [comment]
@Dan33l can you please rebase? The bundler issues got fixed in https://github.com/voxpupuli/puppet-virtualbox/pull/69

File: spec/classes/virtualbox_spec.rb (hunk 1 above)
Comment (Dan33l) [reply spec/classes/virtualbox_spec.rb]: done

File: manifests/init.pp [hunk 2]
```diff
@@ -38,15 +38,16 @@
 class virtualbox (
   String $version               = $virtualbox::params::version,
   String $package_ensure        = $virtualbox::params::package_ensure,
+  String $package_name          = $virtualbox::params::package_name,
```
Comment (bastelfreak) [inline manifests/init.pp]: for which purpose do we add a parameter here?

Comment (bastelfreak) [reply manifests/init.pp]: ah just reordered. Can you undo that?

Comment (Dan33l) [reply manifests/init.pp]: The package have not the same same on all OSes.
Some are `virtualbox` and others `Virtualbox`. This parameter was in master branch and works well.

Comment (Dan33l) [reply manifests/init.pp]: Ah in fact i used the order of the original puppet4 branch.
I reorder to stick with original master.

Comment (Dan33l) [reply manifests/init.pp]: In fact i remember why i follwoed the order in the puppet4 branch. All parameter related to package are grouped. And for me it is better to have all parameters named `package_` grouped together.

**Pull request merged**